### DISCUSSION TITLE
How can I watch data of useLazyAsyncData in server side ?

### DISCUSSION THREAD

**Discussion by teamboot88** [nuxt/nuxt#15970] [Questions] [answered]
https://nuxt.com/docs/api/composables/use-lazy-async-data

When I watch data of useLazyAsyncData,
The watch callback function is not called in server side. (after page refresh)
(In client side, works well)
```js
// pseudo code
const {pending, data} = useLazyAsyncData();

watch (data, newData => {
  // callback (not watched after refresh page)
});
```

How can I watch data of lazy call in server side?
(Should I use no-lazy call?)
Thanks

**Comment by teamboot88** [comment]
https://nuxt.com/docs/api/composables/use-fetch
Should I use "server : false" option?

**Comment by fathi91** [comment, marked as answer]
try add `{ immediate: true }` or use watchEffect 

```javascript
// pseudo code
const {pending, data} = useLazyAsyncData();

watch (data, newData => {
  // callback (not watched after refresh page)
},{ immediate: true });

// or use watchEffect it'll watch any reactive object that used in the callback function
watchEffect(() => {
  data
//....
});
```

**Reply by teamboot88** [reply]
@fathi91 
Thank you! It works like a charm!

```js
watchEffect(() => {
  if (data.value) {
    // 
  }
})
```
